pkg/registry/core/pod/storage/eviction.go

```diff
@@ -220,7 +221,7 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		pdbName = pdb.Name
 
 		// If the pod is not ready, it doesn't count towards healthy and we should not decrement
-		if !podutil.IsPodReady(pod) && pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0 {
+		if !podutil.IsPodReady(pod) && enoughHealthyReplicas(pdb) {
 			updateDeletionOptions = true
 			return nil
 		}
```

Comment:
It seems like after the isReady check, the next logical check is:

```go
if  !disruptionAllowed && disruptionNotAllowedReason == policyv1.SyncFailedReason{
  return fmt.Error("disruption is not allowed because the PDB controller has not recently synchronized the current set of healthy pods")
}
```

If we do not have reliable data to allow a disruption, why would we allow a disruption based on the stale data we do have?

Reply:
This logic is contained in `enoughHealthyReplicas` below.

If you're saying we shouldn't allow any evictions at all (regardless of pod state) if we're in `policyv1.SyncFailedReason` condition, that is what happens today via checkAndDecrement.  The eviction client (eg kubectl) will receive the TooManyRequests error and retry.

Reply:
> , that is what happens today via checkAndDecrement.

Pulling it up to this level would make it more obvious to notice readers like myself.